[2025-10-11T17:46:11.318+0000] {logging_mixin.py:151} INFO - Changing /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:44:03.391375+00:00/task_id=bronze_rig_count_pipeline permission to 509
[2025-10-11T17:46:11.319+0000] {logging_mixin.py:151} INFO - Failed to change /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:44:03.391375+00:00/task_id=bronze_rig_count_pipeline permission to 509: [Errno 1] Operation not permitted: '/opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:44:03.391375+00:00/task_id=bronze_rig_count_pipeline'
[2025-10-11T17:46:11.353+0000] {logging_mixin.py:151} INFO - Changing /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:44:03.391375+00:00/task_id=bronze_rig_count_pipeline permission to 509
[2025-10-11T17:46:11.353+0000] {logging_mixin.py:151} INFO - Failed to change /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:44:03.391375+00:00/task_id=bronze_rig_count_pipeline permission to 509: [Errno 1] Operation not permitted: '/opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:44:03.391375+00:00/task_id=bronze_rig_count_pipeline'
[2025-10-11T17:46:11.404+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ETL_pipeline.bronze_rig_count_pipeline manual__2025-10-11T17:44:03.391375+00:00 [queued]>
[2025-10-11T17:46:11.410+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ETL_pipeline.bronze_rig_count_pipeline manual__2025-10-11T17:44:03.391375+00:00 [queued]>
[2025-10-11T17:46:11.411+0000] {taskinstance.py:1359} INFO - Starting attempt 3 of 3
[2025-10-11T17:46:11.536+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): bronze_rig_count_pipeline> on 2025-10-11 17:44:03.391375+00:00
[2025-10-11T17:46:11.543+0000] {standard_task_runner.py:57} INFO - Started process 465 to run task
[2025-10-11T17:46:11.547+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'ETL_pipeline', 'bronze_rig_count_pipeline', 'manual__2025-10-11T17:44:03.391375+00:00', '--job-id', '7', '--raw', '--subdir', 'DAGS_FOLDER/ETL_pipeline.py', '--cfg-path', '/tmp/tmp4_zw9n75']
[2025-10-11T17:46:11.550+0000] {standard_task_runner.py:85} INFO - Job 7: Subtask bronze_rig_count_pipeline
[2025-10-11T17:46:11.607+0000] {logging_mixin.py:151} INFO - Changing /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:44:03.391375+00:00/task_id=bronze_rig_count_pipeline permission to 509
[2025-10-11T17:46:11.608+0000] {logging_mixin.py:151} INFO - Failed to change /opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:44:03.391375+00:00/task_id=bronze_rig_count_pipeline permission to 509: [Errno 1] Operation not permitted: '/opt/***/logs/dag_id=ETL_pipeline/run_id=manual__2025-10-11T17:44:03.391375+00:00/task_id=bronze_rig_count_pipeline'
[2025-10-11T17:46:11.610+0000] {task_command.py:415} INFO - Running <TaskInstance: ETL_pipeline.bronze_rig_count_pipeline manual__2025-10-11T17:44:03.391375+00:00 [running]> on host 1096c99751cc
[2025-10-11T17:46:11.674+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='bech' AIRFLOW_CTX_DAG_ID='ETL_pipeline' AIRFLOW_CTX_TASK_ID='bronze_rig_count_pipeline' AIRFLOW_CTX_EXECUTION_DATE='2025-10-11T17:44:03.391375+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-10-11T17:44:03.391375+00:00'
[2025-10-11T17:46:11.677+0000] {bronze_layer.py:129} INFO - Starting bronze layer data ingestion...
[2025-10-11T17:46:11.677+0000] {logging_mixin.py:151} INFO - Getting rig count data...
[2025-10-11T17:46:11.678+0000] {bronze_layer.py:50} INFO - Downloading rig count data from https://rigcount.bakerhughes.com/static-files/5844d633-b1f8-45ef-bcb3-fc7a6ac3849b
[2025-10-11T17:46:12.153+0000] {bronze_layer.py:87} ERROR - Failed to read rig count data: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
[2025-10-11T17:46:12.154+0000] {bronze_layer.py:164} ERROR - Bronze layer ingestion failed: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
[2025-10-11T17:46:12.155+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/compat/_optional.py", line 142, in import_optional_dependency
    module = importlib.import_module(name)
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'openpyxl'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/pipeline/bronze_layer.py", line 135, in bronze_etl
    raw_data = get_rig_count_data(RIG_COUNT_URL, False)
  File "/opt/airflow/pipeline/bronze_layer.py", line 72, in get_rig_count_data
    df = pd.read_excel(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_base.py", line 478, in read_excel
    io = ExcelFile(io, storage_options=storage_options, engine=engine)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_base.py", line 1513, in __init__
    self._reader = self._engines[engine](self._io, storage_options=storage_options)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_openpyxl.py", line 548, in __init__
    import_optional_dependency("openpyxl")
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/compat/_optional.py", line 145, in import_optional_dependency
    raise ImportError(msg)
ImportError: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
[2025-10-11T17:46:12.171+0000] {taskinstance.py:1398} INFO - Marking task as FAILED. dag_id=ETL_pipeline, task_id=bronze_rig_count_pipeline, execution_date=20251011T174403, start_date=20251011T174611, end_date=20251011T174612
[2025-10-11T17:46:12.185+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 7 for task bronze_rig_count_pipeline (Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.; 465)
[2025-10-11T17:46:12.243+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-10-11T17:46:12.257+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
